### DISCUSSION TITLE
Artisan 2.4.2 problem with alarm files

### DISCUSSION THREAD

**Discussion by RickGroszkiewicz** [artisan-roaster-scope/artisan#722] [Q&A] [answered]
I thought I was finally ready to start roasting, but OH NO!! When I opened my alarm files to set up a profile for a roast, I noticed that the actions are incorrect. 

In the screen shot below, the action "Slider drum" should instead be "slider burner". See alarms 1, 4, and 9.

The action "Slider burner" should be "slider fan". See alarms 2, 5, 8 and 10:

<img width="822" alt="alarm problem - old Hottop TC4" src="https://user-images.githubusercontent.com/25457554/140250524-82890bc9-155c-41f3-a114-0bfc77fdc6d7.png">

**Comment by RickGroszkiewicz** [comment]
I assume this problem is related to how the devices are configured for my old Hottop with the TC4 (arduino shield). Here are screenshots of my current setup:

Config - Device - Extra devices
<img width="793" alt="Config - Device - Extra devices - old Hottop TC4" src="https://user-images.githubusercontent.com/25457554/140250803-97846e85-9008-416d-a235-64755ca3e076.png">

Config - Events - Config
<img width="825" alt="Config - Events - Config - old Hottop TC4" src="https://user-images.githubusercontent.com/25457554/140250856-e39d8457-e543-43ee-9837-4d2aef94b8bc.png">

Config - Events - Sliders
<img width="823" alt="Config - Events - Sliders - old Hottop TC4" src="https://user-images.githubusercontent.com/25457554/140250883-45b4e79e-2afe-43f5-ba1f-3287a10742d1.png">

NOTE - I can't show you what my artisan setup looked like when I created the alarm files earlier this year. My Linux system got trashed, and I had no viable backups. I had to recreate it from scratch.

**Comment by MAKOMO** [comment, marked as answer]
Rick, Artisan offers 4 event types which can be named freely in the Events dialog. In that dialog on can also define sliders for each of those event types to enter event data and potentially issue some action (like sending a command to an external device, like a roasting machine). An alarm can be defined that sets any of those slider to a defined value if its condition are fulfilled. Obviously your alarm targets do not fit your intended actions. Just correct them.

A note on running Artisan with two different setups that are not compatible (which seems to be the case here for you; as you assigned different event names to the four event types in the two setups you are using): just save the Artisan settings for each setup/machine to a file (menu Help >> Save Settings) and load the corresponding setup for the machine you are going to roast and you should be always fine.

**Reply by RickGroszkiewicz** [reply]
Marko, I will try what you suggested ...

I have maintained two separate alarm files for my two Hottop roasters, for example:
artisan-2.0-settings-Linux-mint-Hottop 2K+.aset
artisan-2.0-settings-Linux-mint-old-Hottop.aset

**Reply by MAKOMO** [reply]
Note that alarm configurations are saved along artisan app settings

**Reply by RickGroszkiewicz** [reply]
I set up my Hottop B-2K+ roaster today. The alarm files look consistent with the older Hottop - thank goodness. 
I will try to Modify the Event sliders as you suggested to get everything working properly again.

**Reply by RickGroszkiewicz** [reply]
After MANY false starts and numerous changes in my artisan settings, I finally completed two roasts on each of my Hottop roasters today. 

I have learned enough that I now plan to get artisan version 2.4.6 working under Linux Mint 20.

**Reply by MAKOMO** [reply]
Excellent Rick! Maybe you can sketch here what you have learned? Thanks!

**Comment by RickGroszkiewicz** [comment]
It has been so long since I set up the artisan software that I was simply doing things based on force of habit. I had forgotten exactly why I did stuff in a particular order - or what NOT to do. Working with two different Hottop models makes it more confusing. 

Here are the things that I re-learned this month

- Do not make changes to artisan settings for a roaster unless it is connected to computer via USB cable
- When loading a profile, and it offers to change extra devices - Just say NO
- I use several "ambient temp" profiles as a starting point
- I had to recreate these profiles for both roasters after I changed the settings so my pre-existing alarm files worked correctly
- Final step before roasting is to load the latest artisan settings file for the current roaster

 
**old Hottop TC4 notes**

Config / Device / ET/BT 
Uncheck the Control box (turns off PID in version 2.4.4)

Config / Events / Sliders
Should not check Coarse, since that limits fan and power settings to multiples of 10%

Config / Curves / Filters
Had to check box for ET <-> BT to swap values


**new Hottop B-2K+ notes**

Config / Device / ET/BT 
Need to check Control box to allow artisan control of roaster

Config / Device / Extra Devices
Uncheck LCD boxes for Hottop heater and fan

Config / Events / Sliders
Should not check Coarse, since that limits fan and power settings to multiples of 10%
